src/tests/multi.test.ts

```diff
@@ -1,51 +1,115 @@
-            }
-
-            return {
-                data: slice.buffer,
-                totalLength: view.byteLength,
-            }
+  it("storing metadata works", async () => {
+    const mockRangeResolver: RangeResolver = async ({
+      start,
+      end,
+      expectedLength,
+    }) => {
+      const bufferSize = 4096 * 2; // 2 blocks of 4kb each
+      const buffer = new ArrayBuffer(bufferSize);
+      const view = new Uint8Array(buffer);
+
+      const metadata = await readBinaryFile("filled_metadata.bin");
+      const metadataLength = metadata.byteLength;
+
+      const dataView = new DataView(buffer);
+      dataView.setUint32(4096 + 24, metadataLength);
+
+      view.set(metadata, 4096 + 24 + 4);
+      const slice = view.slice(start, end + 1);
+
+      if (expectedLength !== undefined && slice.byteLength !== expectedLength) {
+        throw new LengthIntegrityError();
+      }
+
+      return {
+        data: slice.buffer,
+        totalLength: view.byteLength,
+      };
+    };
+
+    const pageFile = new PageFile(mockRangeResolver);
+    const tree = ReadMultiBPTree(mockRangeResolver, pageFile);
+    const metadata = await tree.metadata();
+
+    expect("hello").toEqual(arrayBufferToString(metadata));
+  });
+
+  it("traversing pages", async () => {
+    const mockRangeResolver: RangeResolver = async ({
+      start,
+      end,
+      expectedLength,
+    }) => {
+      const pageSize = 4096;
+      const headerSize = 12;
+      const bufferSize = pageSize * 5; // 1 for gc, 1 for file meta, 3 pages
+      const buffer = new ArrayBuffer(bufferSize);
+      const view = new Uint8Array(buffer);
+      const maxUint64 = BigInt(2) ** BigInt(64) - BigInt(1);
+
+      for (let i = 0; i < 4; i++) {
+        let nextPageOffset;
+        if (i < 3) {
+          nextPageOffset = BigInt((i + 2) * pageSize);
+        } else {
+          nextPageOffset = maxUint64;
         }
-    })
-
 
-    it("storing metadata works", async() => {
+        const offsetPosition = pageSize * (i + 1) + headerSize;
 
-
-
-        const pageFile = new PageFile(mockRangeResolver);
-        const tree = ReadMultiBPTree(mockRangeResolver, pageFile);
-        const metadata = await tree.metadata();
-
-        expect("hello").toEqual(arrayBufferToString(metadata))
-    });
+        if (offsetPosition + 8 <= bufferSize) {
+          const dataView = new DataView(buffer);
+          dataView.setBigUint64(offsetPosition, nextPageOffset);
+        }
+      }
+
+      const slice = view.slice(start, Math.min(end + 1, bufferSize));
+      console.log("slice: ", slice);
+      if (expectedLength !== undefined && slice.byteLength !== expectedLength) {
+        throw new LengthIntegrityError();
+      }
+
+      return {
+        data: slice.buffer,
+        totalLength: view.byteLength,
+      };
+    };
+
+    const pageFile = new PageFile(mockRangeResolver);
+    const tree = ReadMultiBPTree(mockRangeResolver, pageFile);
```

Comment:
Just want to point out that `ReadMultiBPTree` skips the `GC` page and initializes from the `fileMeta`. 

So in this case `tree` is actually the `fileMeta` page